src/plugins/crypto/crypto.c

```diff
@@ -0,0 +1,129 @@
+/**
+* @file
+*
+* @brief filter plugin providing cryptographic operations
+*
+* @copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+*
+*/
+
+#ifndef HAVE_KDBCONFIG
+# include "kdbconfig.h"
+#endif
+#include "crypto.h"
+#include "gcrypt_operations.h"
+
+
+int elektraCryptoOpen(Plugin *handle ELEKTRA_UNUSED, Key *errorKey ELEKTRA_UNUSED)
+{
+	return 1;
+}
+
+int elektraCryptoClose(Plugin *handle ELEKTRA_UNUSED, Key *errorKey ELEKTRA_UNUSED)
+{
+	return 1;
+}
+
+int elektraCryptoGet(Plugin *handle ELEKTRA_UNUSED, KeySet *ks, Key *parentKey)
+{
+	// Publish module configuration to Elektra
+	if (!strcmp (keyName(parentKey), "system/elektra/modules/crypto"))
+	{
+		KeySet *moduleConfig = ksNew (30,
+			keyNew ("system/elektra/modules/crypto", KEY_VALUE, "crypto plugin waits for your orders", KEY_END),
+			keyNew ("system/elektra/modules/crypto/exports", KEY_END),
+			keyNew ("system/elektra/modules/crypto/exports/get", KEY_FUNC, elektraCryptoGet, KEY_END),
+			keyNew ("system/elektra/modules/crypto/exports/set", KEY_FUNC, elektraCryptoSet, KEY_END),
+			keyNew ("system/elektra/modules/crypto/infos/version", KEY_VALUE, PLUGINVERSION, KEY_END),
+			keyNew ("system/elektra/modules/crypto/infos", KEY_VALUE, "Information about crypto plugin is paced in the keys below", KEY_END),
+			keyNew ("system/elektra/modules/crypto/infos/author", KEY_VALUE, "Peter Nirschl <peter.nirschl@gmail.com>", KEY_END),
+			keyNew ("system/elektra/modules/crypto/infos/licence", KEY_VALUE, "BSD", KEY_END),
+			keyNew ("system/elektra/modules/crypto/infos/needs", KEY_VALUE, "", KEY_END),
+			keyNew ("system/elektra/modules/crypto/infos/provides", KEY_VALUE, "", KEY_END),
+			keyNew ("system/elektra/modules/crypto/infos/placements", KEY_VALUE, "postgetstorage presetstorage", KEY_END),
+			keyNew ("system/elektra/modules/crypto/infos/description", KEY_VALUE, "filter plugin for cryptographic operations", KEY_END),
+			KS_END);
+		ksAppend (ks, moduleConfig);
+		ksDel (moduleConfig);
+		return 1;
+	}
+
+	return 1;
+}
+
+int elektraCryptoSet(Plugin *handle ELEKTRA_UNUSED, KeySet *ks ELEKTRA_UNUSED, Key *parentKey ELEKTRA_UNUSED)
+{
+	return 1;
+}
+
+int elektraCryptoError(Plugin *handle ELEKTRA_UNUSED, KeySet *ks ELEKTRA_UNUSED, Key *parentKey ELEKTRA_UNUSED)
+{
+	return 1;
+}
+
+/**
+ * @brief initialize the crypto backend
+ * @retval 1 on success
+ * @retval -1 on failure
+ */
+int elektraCryptoInit()
```

Comment:
This should also happen in elektraCryptoHandleCreate.
Otherwise API is much nicer.